packages/cli-lib/projects.js

```diff
@@ -57,13 +57,14 @@ async function fetchReleaseData(repoName, tag = '') {
 async function downloadProject(
   repoName,
   tag = '',
-  releaseType = GITHUB_RELEASE_TYPES.RELEASE
+  releaseType = GITHUB_RELEASE_TYPES.RELEASE,
+  ref
 ) {
   try {
     let zipUrl;
     if (releaseType === GITHUB_RELEASE_TYPES.REPOSITORY) {
       logger.log(`Fetching ${releaseType} with name ${repoName}...`);
-      zipUrl = `https://api.github.com/repos/HubSpot/${repoName}/zipball`;
+      zipUrl = `https://api.github.com/repos/HubSpot/${repoName}/zipball/${ref}`;
```

Comment:
The github api supports passing in a branch name to specify which branch to download (it defaults to "main" or "master" I think). That means we won't be using any sort of published release when we download the project-compatible version of the boilerplate. Whatever is currently in that branch will get downloaded.

The alternative route is to create a separate repo for the project-compatible version of the boilerplate, with its own release. IMO downloading the branch should be okay, but I figured I should still mention the alternative.

Reply:
I think we should account for `ref = undefined` when building this string. Otherwise, you get a 404 running `hs create api-sample`.

Otherwise, I think this approach works!

Reply:
Good point, updated in fe42874

Reply:
I was looking more into this, and realized there's another option. We could tag the latest commit in the boilerplate branch (w/out linking it to a new release), and then the CLI would download that specific tagged commit. I still think what we have in place now is totally fine since it will be easier to iterate on the boilerplate project. Maybe down the road we could update the flow to use a tagged commit if we're running into issues.